res/controllers/Gemini-G4V-scripts.js

```diff
@@ -0,0 +1,562 @@
+};
+
+// Custom deck definition
+g4v.Deck = function(deckNumbers, midiChannel) {
+    components.Deck.call(this, deckNumbers);
+    this.cueBtn = new components.CueButton([0x8F + midiChannel, 0x02]);
+    this.cupBtn = new 	components.Button({
+        midi: [0x8F + midiChannel, 0x03],
+        inKey: "cue_gotoandplay",
+    });
+    this.playBtn = new components.PlayButton([0x8F + midiChannel, 0x01]);
+    this.syncBtn = new components.SyncButton([0x8F + midiChannel, 0x04]);
+    this.tempoPot = new components.Pot({
+        midi: [0x8F + midiChannel, 0x01],
+        inKey: "rate",
+        shift: function() { this.inKey = "pitch"; },
+        unshift: function() { this.inKey = "rate"; },
+    });
+    this.keyLockBtn = new components.Button({
+        midi: [0x8F + midiChannel, 0x05],
+        key: "keylock",
+        type: components.Button.prototype.types.toggle,
+    });
+    this.quantizeBtn = new components.Button({
+        midi: [0x8F + midiChannel, 0x07],
+        key: "quantize",
+        type: components.Button.prototype.types.toggle,
+    });
+    this.slipBtn = new components.Button({
+        midi: [0x8F + midiChannel, 0x19],
+        key: "slip_enabled",
+        type: components.Button.prototype.types.toggle,
+        deck: this,
+        shift: function() {
+            // TODO - Chenge to next chain when #2618 is merged
+            //this.group = '[EffectRack1_EffectUnit'+script.deckFromGroup(this.deck.currentDeck)+']';
+            //this.inKey = 'next_chain';
+            this.group = "[EffectRack1_EffectUnit"+script.deckFromGroup(this.deck.currentDeck)+"_Effect1]";
+            this.inKey = "next_effect";
+        },
+        unshift: function() {
+            this.group = this.deck.currentDeck;
+            this.inKey = "slip_enable";
+        },
+    });
+    this.fxOnOffBtn = new components.Button({
+        midi: [0x8F + midiChannel, 0x1A],
+        group: "[EffectRack1_EffectUnit"+script.deckFromGroup(this.currentDeck)+"]",
+        key: "group_"+this.currentDeck+"_enable",
+        type: components.Button.prototype.types.toggle,
+    });
+    // Loop Move is a pot not an encoder so requires special treatment
+    this.loopMovePot = new components.Component({
+        key: "loop_move",
+        oldValue: 0,
+        input: function(_channel, _control, value, _status, _group) {
+            this.inSetParameter(value-this.oldValue);
+            this.oldValue = value;
+        },
+    });
+    // Sample Vol is a pot not an encoder so requires special treatment
+    this.sampleVolPot = new components.Component({
+        key: "loop_move",
+        input: function(_channel, _control, value, _status, _group) {
+            for (let i = 1; i < 65; i++) {
+                engine.setValue("[Sampler"+i+"]", "volume", value/0x7f);
+            }
+        },
+    });
+    this.fxSuperPot = new components.Pot({
+        group: "[EffectRack1_EffectUnit"+script.deckFromGroup(this.currentDeck)+"]",
```

Comment:
consider doing `script.deckFromGroup(this.currentDeck)` once in the constructor and then just reusing it. Saves CPU and keeps your code DRYer.